modules/generic_worker/templates/run-generic-worker.sh.erb

```diff
@@ -14,6 +14,39 @@ rm -f tasks-resolved-count.txt
 # wrapper script...
 /usr/local/bin/generic-worker "$@" 2>&1 | logger -t generic-worker -s
 
+exitstatus=( ${PIPESTATUS[0]} )
+
+echo $exitstatus
+
+export TASKCLUSTER_CLIENT_ID="<%= @quarantine_client_id %>"
+export TASKCLUSTER_ACCESS_TOKEN="<%= @quarantine_access_token %>"
+
+if [ $exitstatus -eq 69 ]; then
+    if [ -f /Users/cltbld/quarantined ]; then
+        echo "Worker is quarantined"
+    else
+        echo "Quarantine worker"
+        # quarantine the worker
+        /usr/local/bin/quarantine-worker releng-hardware "<%= @worker_type %>" "<%= @worker_group %>" "<%= @hostname %>" 720h5s
+        # create the bug
+
+        # create a semaphore file
+        echo "quarantined worker" > /Users/cltbld/quarantined
+        # create the bug
+    fi
+elif [ -f /Users/cltbld/quarantined ]; then
+
+    # if the worker was in quarantine, but now the exitcode is other that 69
+    # remove worker from quarantine
+    /usr/local/bin/quarantine-worker releng-hardware "<%= @worker_type %>" "<%= @worker_group %>" "<%= @hostname %>" 0s
```

Comment:
If the worker is in quarantine, it won't be given tasks from the queue, so it will likely exit with exit code 68 (idle timeout) which would not be a reason to take the worker out of quarantine.

I don't think any automated *check* should take the worker out of quarantine. The only way a worker should be taken out of quarantine automatically is if there is e.g. an automatic worker reimage process and there is some way to be sure there is not a hardware failure etc.

We probably also need an override feature so that it is possible to disable the auto-quarantine process on a given worker or set of workers, in case we have a rare intermittent issue that causes a 69 exit code but that is not really a worker failure.

Reply:
+1. This will ease my work. So I suggest to keep auto quarantine and auto generate the bug if the exit code 69 is present, but remove the worker from the quarantine and close the bug will be part of the CI duty task